singularity-eos/closure/mixed_cell_models.hpp

```diff
@@ -219,8 +218,9 @@ class PTESolverBase {
     // material m averaged over the full PTE volume
     rho_total = 0.0;
     for (int m = 0; m < nmat; ++m) {
-      PORTABLE_REQUIRE(vfrac[m] > 0., "Non-positive volume fraction");
-      PORTABLE_REQUIRE(rho[m] > 0., "Non-positive density");
+      PORTABLE_REQUIRE(vfrac[m] > 0.,
+                       "Non-positive volume fraction provided to PTE solver");
+      PORTABLE_REQUIRE(rho[m] > 0., "Non-positive density provided to PTE solver");
```

Comment:
I believe I specifically chose these because we divide by either these or related quantities at some point. In other words, I don't think the PTE solver is capable of handling zero mass/volume materials. I'll look here in a bit to confirm.

I think that's also fine since it's probably appropriate for an extra layer above this to handle non-participating materials. I'd also argue that our `Accessor` wrappers further motivate only sending a subset of materials into the PTE solver and dealing with zeros ahead of time.

Reply:
```c++
rho[m] = robust::ratio(rhobar[m], vfrac[m]);
```
```c++
sie[m] = uscale * robust::ratio(u[m], rhobar[m]);
```
`rhobar[m]` would be zero if `rho[m]` is zero.

Yes we have robust::ratio to handle these, but I'd worry that there could be unintended consequences for dividing by something close to zero. It just seems more appropriate to ask the host code (or an appropriate wrapper) to eliminate materials with zero density or volume fraction.

Reply:
yeah ok fair enough.